### PR TITLE
Feature/feature health info refactor

### PR DESCRIPTION
**Pull request by xavier-campos-diaz**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927&bid=17382707)

File: app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        }
+        dateTime.append(minutes).append(':');
+        if (seconds < FIRST_TWO_DIGITS) {
+            dateTime.append('0');
+        }
+        dateTime.append(seconds);
+        return dateTime.toString();
+    }
+
+    @Override
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java:115]: Codacy Issue found: [Method length is 18 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927)

File: app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        }
+        return DAYS_30;
+    }
+
+    private boolean isLeapYear(int year) {
+        return year % LEAP_YEAR_FREQ == 0 && (year % TWO_LAST_DIGITS != 0
+            || (year / TWO_LAST_DIGITS) % LEAP_YEAR_FREQ == 0);
+    }
+
+    @Override
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java:88]: Codacy Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927)

File: app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    private static final int MIN_END_POS = 16;
+    private static final int SEC_START_POS = 17;
+    private int year;
+    private int month;
+    private int day;
+    private int hour;
+    private int minutes;
+    private int seconds;
+
+    public DateTime(int year, int month, int day, int hour, int minutes, int seconds) throws InvalidFormatException {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java:30]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927)

File: app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        return dateOutOfRange || hourOutOfRange;
+    }
+
+    private boolean isNegative(int year, int month, int hour, int minutes, int seconds) {
+        boolean isYearNegative = year < 0 || month < 0 || day < 0;
+        boolean isDateNegative = hour < 0 || minutes < 0 || seconds < 0;
+        return isYearNegative || isDateNegative;
+    }
+
+    private int numberOfDays(int year, int month) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java:64]: Codacy Issue found: [Method length is 18 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927)

Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java:64]: Codacy Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/UserManagerAdapter.java [hunk 5]
```diff
@@ -16,7 +16,7 @@ public User findUserByUsername(String username) {
         } catch (ExecutionException | InterruptedException e) {
             e.printStackTrace();
         }
-
+        System.out.println("userData: " + userData);
```
Comment (enrichp) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/UserManagerAdapter.java]: print

**Review by enrichp — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetHealthInfo.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927&bid=17382707)

File: app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    public DateTime(String dateTime) {
+        this.year = Integer.parseInt(dateTime.substring(0, LEAP_YEAR_FREQ));
+        this.month = Integer.parseInt(dateTime.substring(MONTH_START_POS, JULY));
+        this.day = Integer.parseInt(dateTime.substring(DAY_START_POS, FIRST_TWO_DIGITS));
+        this.hour = Integer.parseInt(dateTime.substring(HOUR_START_POS, HOUR_END_POS));
+        this.minutes = Integer.parseInt(dateTime.substring(MIN_START_POS, MIN_END_POS));
+        this.seconds = Integer.parseInt(dateTime.substring(SEC_START_POS));
+    }
+
+    private boolean isOutOfRange(int month, int hour, int minutes, int seconds) {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java:60]: You should comment all the methods

File: app/src/main/java/org/pesmypetcare/mypetcare/utilities/InvalidFormatException.java [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package org.pesmypetcare.mypetcare.utilities;
+
+class InvalidFormatException extends Exception {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/utilities/InvalidFormatException.java]: You should use public modifier

**Review by AlbertPG — COMMENTED** [review]
Take a look at my comments

**Review by AlbertPG — APPROVED** [review]
Okey, I see no problem

**Pull request merged**